src/components/firstProfile/location/LocationAutocomplete.tsx

```diff
@@ -1,20 +1,55 @@
-import { useState } from 'react'
+import React, { useState } from 'react'
 import { useSuggestedLocations } from 'hooks/useSuggestedLocations'
-import { Autocomplete, TextField, InputAdornment } from '@mui/material'
-import SearchIcon from '@mui/icons-material/Search'
+import {
+  Autocomplete,
+  TextField,
+  InputAdornment,
+  CircularProgress,
+  Icon,
+} from '@mui/material'
 
 const BOUNCE_DURATION = 500
-const SUGGETIONS_LIMIT = 5
+const SUGGESTIONS_LIMIT = 5
 
-const LocationInputAutocomplete = () => {
+const LocationInputAutocomplete = ({
+  onLocationChange,
+}: {
+  onLocationChange: (location: any) => void
+}) => {
   const [inputLocation, setInputLocation] = useState('')
   const [isFocused, setIsFocused] = useState(false)
+  const [isLoading, setIsLoading] = useState(false) // Local loading state
+
   const suggestedLocations = useSuggestedLocations(
     inputLocation,
     BOUNCE_DURATION,
-    SUGGETIONS_LIMIT
+    SUGGESTIONS_LIMIT
   )
 
+  // Monitor input changes and manage loading state manually
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target
+    setInputLocation(value)
+
+    // Trigger loading state while fetching suggestions
+    if (value.length > 0) {
+      setIsLoading(true)
+      setTimeout(() => {
+        setIsLoading(false) // Simulating loading completion
+      }, BOUNCE_DURATION)
+    } else {
+      setIsLoading(false)
+    }
+  }
+
+  // Handle selected value from Autocomplete
+  const handleSelectLocation = (event: any, value: any) => {
```

Comment:
Please delete `event`(you don't use it anywhere) and specify explicit type for `value`

Reply:
<img width="1018" alt="image" src="https://github.com/user-attachments/assets/264ad098-74e0-4580-b12b-c281300b2d04">

I couldn't make these changes.

Reply:
I see that event is needed
but value isn't a string
![image](https://github.com/user-attachments/assets/f0bd75bf-740e-4100-be29-d3442d53defc)

If it isn't easy to determine the type for `value,` I think we can leave it as `any`.  It looks like the location will need to be changed in the future